iotdb/src/main/java/org/apache/iotdb/db/query/dataset/EngineDataSetWithTimeGenerator.java

```diff
@@ -34,6 +34,7 @@
 
   private EngineTimeGenerator timeGenerator;
   private List<EngineReaderByTimeStamp> readers;
+  private RowRecord cachedRowRecord;
 
```

Comment:
Suggest to add a new flag such as `hasCachedRowRecord` to indicate whether there already has a cached record. Using `cachedRowRecord == null` is not very straightforward.

Not sure whether it's the best way here and we can talk about it.

Reply:
Indeed, it's better to add a hasCachedRowRecord to be the flag instead of cachedRowRecord == null. Fixed.